File: sdkwork-backend-sdk-java/src/main/java/com/sdkwork/backend/api/VoiceSpeakerApi.java
package com.sdkwork.backend.api;

import com.sdkwork.backend.http.HttpClient;
import com.sdkwork.backend.model.*;
import java.util.List;
import java.util.Map;

public class VoiceSpeakerApi {
    private final HttpClient client;
    
    public VoiceSpeakerApi(HttpClient client) {
        this.client = client;
    }

    /** Update an existing voice speaker */
    public PlusApiResultPlusVoiceSpeakerVO update(PlusVoiceSpeakerForm body) throws Exception {
        return (PlusApiResultPlusVoiceSpeakerVO) client.put(ApiPaths.backendPath("/voice/speaker"), body);
    }

    /** Create a new voice speaker */
    public PlusApiResultPlusVoiceSpeakerVO create(PlusVoiceSpeakerForm body) throws Exception {
        return (PlusApiResultPlusVoiceSpeakerVO) client.post(ApiPaths.backendPath("/voice/speaker"), body);
    }

    /** Get voice speakers by page */
    public PlusApiResultPagePlusVoiceSpeakerVO listPublic(PlusVoiceSpeakerQueryForm body, Map<String, Object> params) throws Exception {
        return (PlusApiResultPagePlusVoiceSpeakerVO) client.post(ApiPaths.backendPath("/voice/speaker/list_public"), body, params);
    }

    /** Get voice speakers by page */
    public PlusApiResultPagePlusVoiceSpeakerVO listByPage(PlusVoiceSpeakerQueryForm body, Map<String, Object> params) throws Exception {
        return (PlusApiResultPagePlusVoiceSpeakerVO) client.post(ApiPaths.backendPath("/voice/speaker/list"), body, params);
    }

    /** Get all voice speakers */
    public PlusApiResultListPlusVoiceSpeakerVO listAllEntities(PlusVoiceSpeakerQueryForm body) throws Exception {
        return (PlusApiResultListPlusVoiceSpeakerVO) client.post(ApiPaths.backendPath("/voice/speaker/list/all"), body);
    }

    /** Get a voice speaker by ID */
    public PlusApiResultPlusVoiceSpeakerVO getById(String id) throws Exception {
        return (PlusApiResultPlusVoiceSpeakerVO) client.get(ApiPaths.backendPath("/voice/speaker/" + id + ""));
    }

    /** Delete a voice speaker */
    public PlusApiResultBoolean delete(String id) throws Exception {
        return (PlusApiResultBoolean) client.delete(ApiPaths.backendPath("/voice/speaker/" + id + ""));
    }
}

File: sdkwork-backend-sdk-kotlin/src/main/kotlin/com/sdkwork/backend/api/VoiceSpeakerApi.kt
package com.sdkwork.backend.api

import com.sdkwork.backend.*
import com.sdkwork.backend.http.HttpClient

class VoiceSpeakerApi(private val client: HttpClient) {

    /** Update an existing voice speaker */
    suspend fun update(body: PlusVoiceSpeakerForm): PlusApiResultPlusVoiceSpeakerVO? {
        return client.put(ApiPaths.backendPath("/voice/speaker"), body) as? PlusApiResultPlusVoiceSpeakerVO
    }

    /** Create a new voice speaker */
    suspend fun create(body: PlusVoiceSpeakerForm): PlusApiResultPlusVoiceSpeakerVO? {
        return client.post(ApiPaths.backendPath("/voice/speaker"), body) as? PlusApiResultPlusVoiceSpeakerVO
    }

    /** Get voice speakers by page */
    suspend fun listPublic(body: PlusVoiceSpeakerQueryForm? = null, params: Map<String, Any>? = null): PlusApiResultPagePlusVoiceSpeakerVO? {
        return client.post(ApiPaths.backendPath("/voice/speaker/list_public"), body, params) as? PlusApiResultPagePlusVoiceSpeakerVO
    }

    /** Get voice speakers by page */
    suspend fun listByPage(body: PlusVoiceSpeakerQueryForm? = null, params: Map<String, Any>? = null): PlusApiResultPagePlusVoiceSpeakerVO? {
        return client.post(ApiPaths.backendPath("/voice/speaker/list"), body, params) as? PlusApiResultPagePlusVoiceSpeakerVO
    }

    /** Get all voice speakers */
    suspend fun listAllEntities(body: PlusVoiceSpeakerQueryForm? = null): PlusApiResultListPlusVoiceSpeakerVO? {
        return client.post(ApiPaths.backendPath("/voice/speaker/list/all"), body) as? PlusApiResultListPlusVoiceSpeakerVO
    }

    /** Get a voice speaker by ID */
    suspend fun getById(id: String): PlusApiResultPlusVoiceSpeakerVO? {
        return client.get(ApiPaths.backendPath("/voice/speaker/$id")) as? PlusApiResultPlusVoiceSpeakerVO
    }

    /** Delete a voice speaker */
    suspend fun delete(id: String): PlusApiResultBoolean? {
        return client.delete(ApiPaths.backendPath("/voice/speaker/$id")) as? PlusApiResultBoolean
    }
}
